### PR TITLE
Add Tests to Concat with Plus Waypoint

### LINKED ISSUE #6028: Concatenating Strings with Plus Operator - Allow assignment after definition
From [This Comment](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/5898#issuecomment-170384568)

The second test is over-broad.  It forces you to do assignment on the same line as definition which is not a requirement of the challenge.

This should be valid code:
```js
var myStr;
myStr = "This is the start." + " This is the end.";
```

The test should be broken into two:
1)  See if the + operator has been used, between two strings, regardless of quote type
2)  Other checks to make sure it's assigned to `myStr`, with an appropriate error message, regardless of quote type.

Right now you can make errors not related to using a `+` that say you're not using `+`.

### PR DESCRIPTION
**Pull request by patsul12**
Closes #6028 

I added a few tests here as per @SaintPeter 's issue.

We would now be checking for any two strings being added together using the `+` operator,
a variable name `myStr` being equal to the combination of strings we want,
and that `myStr` is being created somewhere using `var myStr`.

Therefore the code 

```
var myStr;
myStr = "This is the start." + " This is the end.";
```

should now be valid.

### CODE REVIEW

**Comment by pmbenjamin** [comment]
@patsul12 , one of your tests is not accurate enough.

```javascript
"assert(/\\s*var\\s*myStr/.test(code), 'message: <code>myStr</code> should be created using <code>var</code>');",
```
The following **incorrect** code will pass your test because of `\\s*` (which means _match on 0 or more_):
`varmyStr`

```javascript
"assert(/\\s*myStr\\s*=/.test(code), 'message: Make sure to assign the result to a <code>myStr</code> variable.');"
```
This is not an issue, but the first `\\s*` in both tests are unnecessary. See [this regex example](http://regexr.com/3ci1g) I wrote.

**Comment by patsul12** [comment]
Ill fix that first point, overlooked that. However I guess the reason that I put in the \s* before both of them is that the code
`var myStr; myStr = "This is the start." + " This is the end.";
is still valid however ugly, of course if we don't want to accept that I could take it out.

**Comment by pmbenjamin** [comment]
@patsul12 , even in the scenario `var myStr; myStr = "..."+"...";`, the regex `/myStr` would still work correctly. Ref: http://regexr.com/3ci1g

**Comment by patsul12** [comment]
Yeah i see that, brain isn't working too well today i guess haha.

**Comment by pmbenjamin** [comment]
@patsul12 , haha no problem. I have those days too. 
The only thing that's saving me right now is the starbucks coffee that I'm drinking right now!

**Comment by patsul12** [comment]
Yeah, I'm struggling today, already been through a venti coffee lol.

Anywho it should make more sense now i took out the extra \\s* and i changed it to have 
\s{1,} so it requires at least one space between `var` and `myStr`.

**Comment by pmbenjamin** [comment]
`\s+` should do the trick and is a bit more elegant than curly braces.

**Comment by SaintPeter** [comment]
I would prefer the `\s+`, to be more consistent with other challenges.

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 1]
```diff
@@ -920,7 +920,9 @@
       ],
       "tests": [
         "assert(myStr === \"This is the start. This is the end.\", 'message: <code>myStr</code> should have a value of <code>This is the start. This is the end.</code>');",
-        "assert(/var\\s+myStr\\s*=\\s*([\"'])This is the start\\. \\1\\s*\\+\\s*([\"'])This is the end\\.([\"'])/.test(code), 'message: Use the <code>+</code> operator to build <code>myStr</code>');"
+        "assert(/([\"']).*([\"'])\\s*\\+\\s*([\"']).*([\"'])/.test(code), 'message: Use the <code>+</code> operator to build <code>myStr</code>');",
+        "assert(/var\\s{1,}myStr/.test(code), 'message: <code>myStr</code> should be created using <code>var</code>');",
+        "assert(/myStr\\s*=/.test(code), 'message: Make sure to assign the result to a <code>myStr</code> variable.');"
```
Comment (SaintPeter) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: to "the" myStr variable.  There should only be one.

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 2]
```diff
@@ -920,7 +920,9 @@
       ],
       "tests": [
         "assert(myStr === \"This is the start. This is the end.\", 'message: <code>myStr</code> should have a value of <code>This is the start. This is the end.</code>');",
-        "assert(/var\\s+myStr\\s*=\\s*([\"'])This is the start\\. \\1\\s*\\+\\s*([\"'])This is the end\\.([\"'])/.test(code), 'message: Use the <code>+</code> operator to build <code>myStr</code>');"
+        "assert(/([\"']).*([\"'])\\s*\\+\\s*([\"']).*([\"'])/.test(code), 'message: Use the <code>+</code> operator to build <code>myStr</code>');",
+        "assert(/var\\s{1,}myStr/.test(code), 'message: <code>myStr</code> should be created using <code>var</code>');",
```
Comment (SaintPeter) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: using "the <code>var</code> keyword"

**Comment by patsul12** [comment]
Ah I see,  on the train right now I'll fix both when I have a moment!

**Comment by patsul12** [comment]
Fixed! should be good to go (i hope haha.)

**Comment by SaintPeter** [comment]
This code passes the operator presence test:
```
// Example
var ourStr = "I come first. " + "I come second.";

// Only change code below this line
var myStr;
```
You already have one above, for `outStr`.
You can do:
```
assert(code.match(<regex>).length > 1, // etc
```

**Comment by patsul12** [comment]
changed it to a .match expression

**Comment by SaintPeter** [comment]
Excellent.  Thank you!

**Pull request merged**